lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,44 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = formatted_cause(exception)
+
+          lines += potential_lines unless potential_lines.empty?
+
+          lines
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            exceptions = []
+
+            while exception.respond_to?(:cause) && exception.cause
```

Comment:
Thank you I think I prefer using test_doubles of the `Exception` class in this case as I think that looks nicer, that's only if you have no objections to me doing that :+1: